warden/x/intent/keeper/query_simulate_intent.go

```diff
@@ -0,0 +1,33 @@
+package keeper
+
+import (
+	"context"
+
+	// sdk "github.com/cosmos/cosmos-sdk/types"
+	// "github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/warden-protocol/wardenprotocol/shield"
+	"github.com/warden-protocol/wardenprotocol/shield/object"
+	"github.com/warden-protocol/wardenprotocol/warden/x/intent/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) SimulateIntent(goCtx context.Context, req *types.QuerySimulateIntentRequest) (*types.QuerySimulateIntentResponse, error) {
+	// ctx := sdk.UnwrapSDKContext(goCtx)
+
+	expr, err := shield.Parse(req.Definition)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "Failed to parse input definition")
```

Comment:
can we add the `err` value in the error string so we give a hint of what's wrong?